### DISCUSSION TITLE
Connecting directus database to postgres 12

### DISCUSSION THREAD

**Discussion by kaiquefeng** [directus/directus#12234] [Archive]
I'm trying to connect AWS RDS free postgress database postgres 12 but I'm getting a syntax error.

I'm using knex to write the database tables.

I have a field in the table that I use gen_random_uuid() to generate the uuid and it doesn't recognize it

![image](https://user-images.githubusercontent.com/56850504/159069725-e2c4de4c-b48e-498a-9b1b-1966e39c056e.png)

**Comment by rijkvanzanten** [comment]
Are you using Directus to create this table?

**Reply by kaiquefeng** [reply]
I'm using docker directus to create tables with knex

**Reply by kaiquefeng** [reply]
I am copyng archives knex to folder directus migrations

**Reply by kaiquefeng** [reply]
![image](https://user-images.githubusercontent.com/56850504/159070758-192f075e-3350-4926-9da1-e42de57c1c81.png)

**Reply by rijkvanzanten** [reply]
OK. Keep in mind that Directus does not use Knex migrations in its custom migrations. Directus itself doesn't use `gen_random_uuid()`, so I think you'll have to update your custom migration to not use that 👍🏻

**Reply by kaiquefeng** [reply]
How can directus migrations generate a uuid automatically?

**Reply by rijkvanzanten** [reply]
I don't know what you're trying right to do necessarily, so I don't really know how to answer that..

**Reply by kaiquefeng** [reply]
when creating an item from a collection it generates a uuid for each item, but I already want to make it ready via migration

**Reply by kaiquefeng** [reply]
![image](https://user-images.githubusercontent.com/56850504/159072523-390cac33-6452-4877-b81e-79a34a233369.png)

I can create a migration that already pre-set the use of ( generate ans Save uui )?

**Comment by kaiquefeng** [comment]
Resolved!

I using flag ( special ) for set uuid or other defaults

![image](https://user-images.githubusercontent.com/56850504/159078728-82b9181b-6daa-40a3-832a-4c2f92b0af59.png)
